tests/rest/client/test_sync.py

```diff
@@ -417,7 +419,23 @@ def test_hidden_read_receipts(self):
         # Test that the first user can't see the other user's hidden read receipt
         self.assertEqual(self._get_read_receipt(), None)
 
-    def test_read_receipt_with_empty_body(self):
+    @parameterized.expand(
+        [
+            (
+                "agent1",
+                "Element/1.2.2 (Linux; U; Android 9; MatrixAndroidSDK_X 0.0.1)",
+                200,
+            ),
+            ("agent2", "Element/1.2.2 (Linux; Android 6)", 200),
+            ("agent3", "SchildiChat/1.3.6 (Android 11)", 400),  # Will allow empty body again starting at version 1.3+
+            ("agent4", "Element (Riot.im) (Android 9)", 200),
+            ("agent5", "Element/1.2.1", 400),  # Does not contain "Android"
+            ("agent6", "Element dbg/1.1.8-dev (Android)", 400),  # Different format, missing "/" after Element
```

Comment:
I think we're missing a test that old SchildiChat versions will be exempt from spec compliance.
```suggestion
            ("agent6", "Element dbg/1.1.8-dev (Android)", 400),  # Different format, missing "/" after Element
            ("agent7", "SchildiChat/1.2.1 (Android)", 200), 
```

Reply:
Added.